samples/python/reactors/PorousMediaBurner.py

```diff
@@ -0,0 +1,395 @@
+            RHS[self.index_Ts] += heat_flux(self, self.neighbor_right) * self.A
+
+        # ============================================================================#
+        #                             species mass fractions                          #
+        # ============================================================================#
+
+        wdot = self.kinetics.net_production_rates # chemical source terms
+        if not self.chemistry:
+            wdot *= 0.0
+
+        # right hand side of the mass fraction equations
+        for k in range(self.thermo.n_species):
+            index = self.component_index(self.thermo.species_name(k))
+            # chemical source term
+            RHS[index] += wdot[k] * self.thermo.molecular_weights[k] / density
+            # convective contribution
+            RHS[index] += self.A * mdot / \
+                (self.V * porosity * density) * (Y_in[k] - self.thermo.Y[k])
+
+        # ============================================================================#
+        #                             Temperature of the gas                          #
+        # ============================================================================#
+
+        # right hand side of the gas phase temperature equation
+        enthalpy_loss = np.dot(self.thermo.partial_molar_enthalpies
+                               / self.thermo.molecular_weights, Y_in)
+
+        hrr = -np.dot(self.thermo.partial_molar_enthalpies, wdot)  # (W/m^3)
+
+        Tindex = self.component_index("temperature")
+        LHS[Tindex] = porosity * density * cp * self.V
+        # heat transfer between solid and gas-phase
+        RHS[Tindex] -= self.htc() * (self.thermo.T - self.Ts) * self.V
+        # convective transport
+        RHS[Tindex] += self.A * mdot * (h_in - enthalpy_loss)
+        # chemical contribution
+        RHS[Tindex] += hrr * self.V
+
+    def before_component_index(self, name):
+        if name == "Ts":
+            return self.index_Ts
+
+    def before_component_name(self, i):
+        if i == self.index_Ts:
+            return "Ts"
+
+
+# list of reactors that form the cascade
+reactors = []
+
+# Define 7 reactors, each with a different length (m), and total length of 4 inches.
+# Increasing the number of reactors improves the "resolution" and thus the accuracy
+lengths = [
+    1.5 * inch,
+    0.25 * inch,
+    0.25 * inch,
+    0.125 * inch,
+    0.125 * inch,
+    0.75 * inch,
+    1 * inch]
+
+total_length = np.sum(lengths)
+
+sum_lengths = 0.
+# add each reactor one by one to the list
+for rL in lengths:
+
+    midpoint = sum_lengths + 0.5 * rL  # midpoint of the current reactor
+    sum_lengths += rL
+
+    if midpoint < total_length * 0.5:
+        # for the left half of the burner, fill all reactors with cold inert 40 PPI YZA
+        props = ReactorProperties(
+            diameter=2 * inch,
+            length=rL,
+            midpoint=midpoint,
+            chemistry=False,
```

Comment:
The description of the example should mention that the upstream section does not consider chemistry.